### PR TITLE
 Added Proxy App AMG 2013

### PR DESCRIPTION
**Pull request by akhilpatlolla**

### CODE REVIEW

File: var/spack/repos/builtin/packages/amg/package.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from spack import *
+
+
+class Amg(MakefilePackage):
+    """ AMG2013 is a parallel algebraic multigrid solver for linear
+        systems arising from problems on unstructured grids.
+        It has been derived directly from the BoomerAMG solver in the
+        hypre library, a large linear solver library that is being developed
+        in the Center for Applied Scientific Computing (CASC) at LLNL.
+    """
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/amg/package.py]: I don't think the doc tests will like this docstring. I always format multi-line docs like so:
```python
"""Here is a multi-line docstring
that stretches onto multiple lines.
"""
```

File: var/spack/repos/builtin/packages/amg/package.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    tags = ['proxy-app']
+    homepage = "https://codesign.llnl.gov/amg2013.php"
+    url      = "https://codesign.llnl.gov/amg2013/amg2013.tgz"
+
+    version('2013', '9d918d2a69528b83e6e0aba6ba601fef')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/amg/package.py]: It seems like the package is referred to as `amg2013`, not `amg`. Thus, `2013` isn't really the version number. I don't seem to be able to find individual downloads for each version, so maybe we should rename the package to `amg2013` and change the version to `release` or `master` or something like that?

File: var/spack/repos/builtin/packages/amg/package.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        include_lflags = '-lm '
+
+        if self.compiler.name == 'intel':
+            include_lflags += '-qsmp '
+
+        if '+mpi' in self.spec:
+            if '+openmp' in self.spec:
+                include_cflags += '-DHYPRE_USING_OPENMP' + ' '
+            if '+assumepartition' in self.spec:
+                include_cflags += '-DHYPRE_NO_GLOBAL_PARTITION' + ' '
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/amg/package.py]: Does this mean that it's not possible to use `+openmp` or `+assumepartition` with `-mpi`? If so you should make this explicit above:
```python
conflicts('+openmp', when='~mpi')
conflicts('+assumepartition', when='~mpi')
```

File: var/spack/repos/builtin/packages/amg/package.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+            include_cflags += '-DTIMER_USE_MPI ' + self.compiler.openmp_flag
+
+            include_lflags += ' ' + self.compiler.openmp_flag
+            targets.append('INCLUDE_CFLAGS={0}'.format(include_cflags))
+            targets.append('INCLUDE_LFLAGS={0}'.format(include_lflags))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/amg/package.py]: You don't want to use any flags for `~mpi`?

**Comment by adamjstewart** [comment]
P.S. You don't need to close the PR or delete the branch if you plan on making changes to it. You can just push new commits to the same branch.

**Comment by akhilpatlolla** [comment]
I had bad merge issue on this branch so I removed it !

**Pull request closed without merging**